### PR TITLE
Change how qthreads gets hwloc topology from the Chapel runtime

### PR DESCRIPTION
**Pull request by jhh67**
`Qthreads` and the Chapel runtime must share the same `hwloc` topology to ensure that there are no inconsistencies between how they view the hardware. Previously, this was accomplished via the `hwloc_via_chapel` `qthreads` "topology interface" which was a copy of the `hwloc` interface, modified to get the `hwloc` topology from the Chapel runtime. This PR adds a `--with-hwloc-get-topology-function` option to `qthreads` `configure` that specifies a function to be called to get the `hwloc` topology, and the `hwloc` and `binders` interfaces have been modified to call it.

Passes all single-locale tests.

### CODE REVIEW

File: third-party/qthread/Makefile [hunk 1]
```diff
@@ -158,6 +157,10 @@ ifeq ($(CHPL_LIB_PIC),pic)
 CFLAGS += $(SHARED_LIB_CFLAGS)
 endif
 
+ifneq (, $(CHPL_QTHREAD_DEBUG))
+    CHPL_QTHREAD_CFG_OPTIONS += --enable-debugging --enable-debug=$(CHPL_QTHREAD_DEBUG)
```
Comment (ronawho) [inline third-party/qthread/Makefile:161]: I'm fine with adding this, but note that you could have achieved this before with `CHPL_QTHREAD_MORE_CFG_OPTIONS="--enable-debugging --enable-debug=<debug options>"` (and most third-parties support this env var to add arbitrary configure options)

File: third-party/qthread/qthread-src/src/affinity/binders.c [hunk 2]
```diff
@@ -91,9 +91,15 @@ int INTERNAL qt_affinity_compact(int num_workers,
 void INTERNAL qt_affinity_init(qthread_shepherd_id_t *nbshepherds,
                                qthread_worker_id_t   *nbworkers,
                                size_t                *hw_par)
-{                           
-  hwloc_topology_init(&topology);
-  hwloc_topology_load(topology);
+{                      
+#ifdef HWLOC_GET_TOPOLOGY_FUNCTION
+  extern hwloc_topology_t HWLOC_GET_TOPOLOGY_FUNCTION;
```
Comment (ronawho) [inline third-party/qthread/qthread-src/src/affinity/binders.c:96]: I think technically `chpl_topo_getHwlocTopology` returns `void*`. I don't think the different declared types will hurt anything, but wanted to check that you agree.

**Review by ronawho — COMMENTED** [review]
Left some questions inline, but overall looks great. Please:
 - Update the qthread README to note this patch. This reminds us of what patches we have to apply when upgrading or which ones we should try to upstream and is a trigger for the build system that qthreads needs to be rebuilt (although updating the Makefile also does that)
 - create a private issue to remind us to upstream this change

File: third-party/qthread/Makefile (hunk 1 above)
Comment (jhh67) [reply third-party/qthread/Makefile]: Removed this.

File: third-party/qthread/qthread-src/src/affinity/binders.c (hunk 2 above)
Comment (jhh67) [reply third-party/qthread/qthread-src/src/affinity/binders.c]: Fixed.

**Comment by jhh67** [comment]
Updated the README and created the private issue.

File: third-party/qthread/qthread-src/src/affinity/binders.c [hunk 3]
```diff
@@ -1,11 +1,11 @@
-#include <hwloc.h>
+/ex#include <hwloc.h>
```
Comment (ronawho) [inline third-party/qthread/qthread-src/src/affinity/binders.c:1]: accident?

**Review by ronawho — APPROVED** [review]
👍

File: third-party/qthread/qthread-src/src/affinity/binders.c (hunk 3 above)
Comment (jhh67) [reply third-party/qthread/qthread-src/src/affinity/binders.c]: Fixed.

**Comment by jhh67** [comment]
I've make the requested changes, created the private issue, and verified that the single-locale tests pass with both `hwloc` and `binders` as the `qthread` topology interface.

**Review by ronawho — APPROVED** [review]

**Pull request merged**